tests/dedicatedevhtests/l7_evh_crd_test.go

```diff
@@ -0,0 +1,440 @@
+	SetUpIngressForCacheSyncCheck(t, false, false, modelName)
+
+	err := integrationtest.SetUpOAuthSecret()
+	if err != nil {
+		t.Fatalf("error in creating my-oauth-secret: %v", err)
+	} else {
+		// Sleeping for 5s for secret to be updated in informer
+		time.Sleep(5 * time.Second)
+	}
+	integrationtest.SetupSSORule(t, srname, "foo.com", "SAML")
+
+	g.Eventually(func() string {
+		ssoRule, _ := v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Get(context.TODO(), srname, metav1.GetOptions{})
+		return ssoRule.Status.Status
+	}, 20*time.Second).Should(gomega.Equal("Accepted"))
+
+	sniVSKey := cache.NamespaceName{Namespace: "admin", Name: lib.Encode("cluster--foo.com"+lib.DedicatedSuffix, lib.EVHVS) + "-EVH"}
+	integrationtest.VerifyMetadataSSORule(t, g, sniVSKey, "default/samplesr-foo", true)
+	g.Eventually(func() int {
+		_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		nodes, ok := aviModel.(*avinodes.AviObjectGraph)
+		if !ok {
+			return 0
+		}
+		return len(nodes.GetAviEvhVS())
+	}, 20*time.Second).Should(gomega.Equal(1))
+
+	_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+	nodes := aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+
+	g.Expect(nodes[0].OauthVsConfig).To(gomega.BeNil())
+	g.Expect(*nodes[0].SsoPolicyRef).To(gomega.ContainSubstring("thisisaviref-ssopolicysaml"))
+	g.Expect(nodes[0].SamlSpConfig.AcsIndex).To(gomega.BeNil())
+	g.Expect(*nodes[0].SamlSpConfig.AuthnReqAcsType).To(gomega.Equal("SAML_AUTHN_REQ_ACS_TYPE_NONE"))
+	g.Expect(*nodes[0].SamlSpConfig.CookieName).To(gomega.Equal("MY_SAML_COOKIE"))
+	g.Expect(*nodes[0].SamlSpConfig.CookieTimeout).To(gomega.Equal(int32(120)))
+	g.Expect(*nodes[0].SamlSpConfig.EntityID).To(gomega.Equal("my-entityid"))
+	g.Expect(*nodes[0].SamlSpConfig.SigningSslKeyAndCertificateRef).To(gomega.ContainSubstring("thisisaviref-sslkeyandcertrefsaml"))
+	g.Expect(*nodes[0].SamlSpConfig.SingleSignonURL).To(gomega.Equal("https://auth.com/sso/acs/"))
+	g.Expect(*nodes[0].SamlSpConfig.UseIdpSessionTimeout).To(gomega.Equal(false))
+
+	//Update with oauth parameters instead of saml
+	srUpdate := integrationtest.FakeSSORule{
+		Name:      srname,
+		Namespace: "default",
+		Fqdn:      "foo.com",
+		SSOType:   "OAuth",
+	}.SSORule()
+	srUpdate.ResourceVersion = "2"
+	_, err = v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Update(context.TODO(), srUpdate, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating SSORule: %v", err)
+	}
+	g.Eventually(func() string {
+		ssoRule, _ := v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Get(context.TODO(), srname, metav1.GetOptions{})
+		return ssoRule.Status.Status
+	}, 10*time.Second).Should(gomega.Equal("Accepted"))
+
+	g.Eventually(func() int {
+		_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		nodes, ok := aviModel.(*avinodes.AviObjectGraph)
+		if !ok {
+			return 0
+		}
+		return len(nodes.GetAviEvhVS())
+	}, 20*time.Second).Should(gomega.Equal(1))
```

Comment:
same here. This code is not required since the model will be already present.

Reply:
removed